lib/catalog.ex

```diff
@@ -55,26 +55,17 @@ defmodule Catalog do
   @spec handle_continue({:load_documents, keyword()}, t()) :: {:noreply, t()}
   def handle_continue({:load_documents, opts}, %__MODULE__{ets: ets} = state) do
     documents_fn = Keyword.get(opts, :documents_fn, &all_documents/0)
+    documents = documents_fn.()
 
-    start_time = Time.utc_now()
     Logger.info(fn -> "Loading document metadata" end)
 
-    for document <- documents_fn.() do
-      _ =
-        :ets.insert(
-          ets,
-          {Document.key(document), document.last_name, document.first_name,
-           document.date_of_birth, document}
-        )
-    end
+    {time_in_µs, _} = :timer.tc(&populate_ets_table/2, [ets, documents])
 
     # Let the health server know that the data has finished loading
     Catalog.Health.loaded()
 
     Logger.info(fn ->
-      "Finished loading document metadata, time_in_ms=#{
-        Time.diff(Time.utc_now(), start_time, :millisecond)
-      }"
+      "Finished loading document metadata, time_in_ms=#{time_in_µs / 1_000}"
```

Comment:
FWIW, this is now slightly different: the previous behavior included the time of `document_fn` in the reported time. Not sure if that's what you want or not.

Reply:
Oh good catch! I definitely do want to include that.